rpmlint/checks/IconSizesCheck.py

```diff
@@ -0,0 +1,29 @@
+import re
+
+from rpmlint.checks.AbstractCheck import AbstractCheck
+
+
+class IconSizesCheck(AbstractCheck):
+    def __init__(self, config, output):
+        super().__init__(config, output)
+        self.file_size_regex = re.compile(r'/icons/[^/]+/(?P<x>\d+)x(?P<y>\d+)/')
+        self.info_size_regex = re.compile(r'(?P<x>\d+) x (?P<y>\d+)')
+
+    def check(self, pkg):
+        if pkg.isSource():
+            return
+
+        for fname, pkgfile in pkg.files().items():
+            if '/animations/' in fname:
+                continue
+
+            res = self.file_size_regex.search(fname)
```

Comment:
This does not need to be so nested, just get both of the 'res' values at once and then if both contain results do the comparisons, no need to do if, anther if, another if....